registry/scheduledTasks.xml

```diff
@@ -31,4 +31,8 @@
 		<descr>Send automated reminders to reviewers to complete their assignments.</descr>
 		<frequency hour="0"/>
 	</task>
+	<task class="lib.pkp.classes.task.StatisticsReport">
+		<descr>Send automated statistics reports to journal managers and editors.</descr>
+		<frequency day="1"/>
```

Comment:
@NateWr, correct me if I'm wrong, but in my opinion this is what caused the spam =]
I imagine some users scheduled the task to run in a smaller period of time and the `day="1"` doesn't block consecutive executions.

When using `day`, [this](https://github.com/pkp/pkp-lib/blob/master/classes/scheduledTask/ScheduledTaskHelper.inc.php#L234) will check if one month has passed, but then it will be overwritten/ignored [here](https://github.com/pkp/pkp-lib/blob/master/classes/scheduledTask/ScheduledTaskHelper.inc.php#L246). Thus this task will be accepted every time the scheduler gets executed on day 1.

The only way I found to make it work with the current code is by using something like `day="1-1/1"`.

ps: In my opinion we'd better replace this piece of code by a cron parser.

Reply:
Thanks Jonas. @asmecher you're more familiar with this code than me. Does this look like a bug when scheduling a monthly frequency or should it be updated to `1-1/1`?

Reply:
Ahh, interesting. And I note that our recommendation for cron jobs (`docs/README.md`) is that they execute hourly. I suppose we haven't encountered this problem before because all the tasks I can think of have their own "did I already do this" check, e.g. review reminders.

That's a bug in the scheduler, I think. It's hard to argue that `day="1"` should mean "run as often as you like, as long as it's only on the 1st of the month". The problem behavior IMO is that the check on the last run time gets overridden so easily.

Reply:
@asmecher Exactly, what do you think about replacing it by the cron parser? I'm using such things to schedule jobs that deal with money and it has been working well for the customers (I've added a cron editor component).

Reply:
I've filed this as an issue against the 3.2 milestone: https://github.com/pkp/pkp-lib/issues/5362

Reply:
@jonasraoni, do you mean a 3rd-party PHP implementation of a CRON parser, or something different?

Reply:
The former. I think it's easier to forward this task to a well established format, even tough the accuracy will be limited by the frequency that the users schedule the job.

Reply:
I would be very supportive of removing our hand-made solution here in favour of something 3rd-party that's available e.g. via Composer. Just keeping in mind we need to support two means of invocation: Cron-based (or equivalent), and something piggybacking onto a web request (the Acron plugin, or something like it).